src/main/java/edu/kis/powp/jobs2d/drivers/adapter/BasicLineDrawerAdapter.java

```diff
@@ -23,11 +25,12 @@ public void setPosition(int x, int y) {
 
 	@Override
 	public void operateTo(int x, int y) {
-		ILine line = LineFactory.getBasicLine();
+		ILine line = new ParametrizedLineAdapter(LineFactory.getBasicLine());
 		line.setStartCoordinates(this.startX, this.startY);
 		line.setEndCoordinates(x, y);
 
-		drawLine(line);
+		setPosition(x, y);
+		DrawerFeature.getDrawerController().drawLine(line);
```

Comment:
Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.